cypress/e2e/embedded_superset_dashboard.cy.js

```diff
@@ -0,0 +1,210 @@
+        cy.contains('Configure source').should('be.visible').click()
+
+        // A modal form to create a new embedded dashboard is showing
+        cy.contains(
+            'New dashboard: configure external source (superset)'
+        ).should('be.visible')
+
+        // Check all initial values and change them
+        getInputByLabelText('Title').should('have.value', '').type(NAME)
+        getInputByLabelText('Code').should('have.value', '').type(CODE)
+        getInputByLabelText('Description', 'textarea')
+            .should('have.value', '')
+            .type(DESCRIPTION)
+        getInputByLabelText('Superset Embed ID')
+            .should('have.value', '')
+            .type(UUID)
+        getInputByLabelText('Show chart controls on dashboard items')
+            .should('be.checked')
+            .uncheck()
+        getInputByLabelText('Expand filters').should('not.be.checked').check()
+
+        // Click the create button
+        cy.contains('Save dashboard').should('be.enabled').click()
+
+        cy.contains('h3', NAME).should('be.visible')
+        cy.contains('External data').should('be.visible')
+        // An iframe should be visible with the UUID in the src
+        cy.get('iframe')
+            .should('be.visible')
+            .and('have.attr', 'src')
+            .and('contain', UUID)
+    })
+
+    it('has some options disabled in the action-bar', () => {
+        // Primary actions
+        cy.contains('button', 'Slideshow').should('be.disabled')
+        cy.contains('button', 'Filter').should('be.disabled')
+        // Actions in the more-menu
+        cy.getByDataTest('more-actions-button').should('be.enabled').click()
+        cy.contains('a', 'Make available offline').should(
+            'have.attr',
+            'aria-disabled',
+            'true'
+        )
+        cy.contains('a', 'Print')
+            .should('have.attr', 'aria-disabled', 'true')
+            .and('have.attr', 'aria-expanded', 'false')
+        // Close the menu by clicking the backdrop
+        cy.get('.backdrop').should('be.visible').click()
+    })
+
+    it('shows the description', () => {
+        cy.getByDataTest('more-actions-button').should('be.enabled').click()
+        cy.contains('a', 'Show description').should('be.visible').click()
+        cy.contains(DESCRIPTION).should('be.visible')
+        // Keep visible so we can check it updates correctly later on
+    })
+
+    it('stars and unstars the superset embedded dashboard', () => {
+        // Can be starred via menu bar
+        cy.getByDataTest('dashboard-unstarred').click()
+        cy.getByDataTest('dashboard-starred').should('be.visible')
+        // And unstarred via ...menu
+        cy.getByDataTest('more-actions-button').should('be.enabled').click()
+        cy.contains('a', 'Unstar dashboard').should('be.visible').click()
+        cy.getByDataTest('dashboard-unstarred').should('be.visible')
+    })
+
+    it('can share the superset embedded dashboard', () => {
```

Comment:
rename test to reflect that it is not actually sharing. i.e. "can open the sharing dialog"